118 function to fix incorrect guids

Heaghneyn opened the pull request:
to test take one of the afp linelists from POLIS_test/data/Core_Ready_Files/ (this fix is only for cases 2020 onward) and look at cases where geo.corrected = 1. admin0/1/2guid is the original guid from POLIS, adm0/1/2guid is the reformatted and corrected version. cases with geo.corrected = 1 will have a difference in at least one of the guids

sample code to check:
 
check.afp <- `afp_linelist_2019-01-01_2025-01-30` |>
  filter((paste0("{", stringr::str_to_upper(admin2guid), "}", sep = "") != adm2guid |
         paste0("{", stringr::str_to_upper(admin1guid), "}", sep = "") != adm1guid |
         paste0("{", stringr::str_to_upper(admin0guid), "}", sep = "") != adm0guid) &
         geo.corrected == 0) |>
  select(epid, yronset, place.admin.0, place.admin.1, place.admin.2, admin0guid, admin1guid, admin2guid, adm0guid, adm1guid, adm2guid, geo.corrected)

ensure that all cases with differences in POLIS vs corrected GUIDs have the geo.corrected flag. you can then take a sample of those cases and plot with the global district shape files to make sure that they plot in the designated shape.

mcuadera reviewed (changes requested):
Reviewed the changes and confirmed that the cases are being mapped to the appropriate corrected GUIDs. A couple of questions/suggestions:
1. This branch seems to have been pulled from `master`. Cross-referenced the DESCRIPTION file of `master` and `dev`. There could potentially be merge conflicts in this case. There are warnings such as: 
```"Warning: replacing previous import ‘dplyr::filter’ by ‘stats::filter’ when loading ‘tidypolis’"``` that we probably want to avoid.
3. While the GUIDs are corrected, I noted some records in `place.admin.[0-2]` columns are still `NA`s. It would be good to potentially fill this in. Additionally, to perform a check for non-`NA` records and ensure what is put in as the `place.admin.[0-2]` matches what we have in the geodatabase. I think you tried to do some filling in of `NA`s on lines 1374-1382 of `utils.R`, but something isn't working. It's hard for me to check as `ADM[0-2]_NAME` don't exist in `afp_linelist_2020-01-01_2025-01-30.rds`. 
4. Run a `check()` and ensure no errors/warnings exist. I believe most of the errors have been resolved from `main` in our `dev` branch. This will also solve some issues associated with point 1 above.
5. Update the `renv` snapshot if you've recently updated any packages.

Heaghneyn commented:
rebased from dev

I've fixed the issue with names, there remain 216 cases where I've not been able to fill names, there seem to be 2 reasons for this:
1. some countries, particularly EURO countries don't have districts that in the global.dist shape files (Albania, Macedonia) but do have shapes in country and province. 
2. example in Mozambique: there are cases that have valid country and province GUIDs but are blank in district, the lat/lon don't join to the shapes and don't have a district for which to place random points. I could read in province and country and alternatively place random points based on the lowest level of geographical info we have but not sure if worth it for 216 cases where we'd be randomly bouncing cases from district to district each week (or province to province if the lowest admin level is country)
